public/app/features/datasources/components/DataSourcesListCard.tsx

```diff
@@ -0,0 +1,128 @@
+  hasWriteRights: boolean;
+  hasExploreRights: boolean;
+}
+
+export function DataSourcesListCard({ dataSource, hasWriteRights, hasExploreRights }: Props) {
+  const dataSourcesRoutes = useDataSourcesRoutes();
+  const dsLink = config.appSubUrl + dataSourcesRoutes.Edit.replace(/:uid/gi, dataSource.uid);
+  const styles = useStyles2(getStyles);
+
+  return (
+    <Card href={hasWriteRights ? dsLink : undefined}>
+      <Card.Heading>{dataSource.name}</Card.Heading>
+      <Card.Figure>
+        <img src={dataSource.typeLogoUrl} alt="" height="40px" width="40px" className={styles.logo} />
+      </Card.Figure>
+      <Card.Meta>
+        {[
+          dataSource.typeName,
+          dataSource.url,
+          dataSource.isDefault && <Tag key="default-tag" name={'default'} colorIndex={1} />,
+        ]}
+      </Card.Meta>
+      <Card.Tags>
+        {/* Build Dashboard */}
+        <LinkButton
+          icon="apps"
+          fill="outline"
+          variant="secondary"
+          href={`dashboard/new-with-ds/${dataSource.uid}`}
+          onClick={() => {
+            trackCreateDashboardClicked({
+              grafana_version: config.buildInfo.version,
+              datasource_uid: dataSource.uid,
+              plugin_name: dataSource.typeName,
+              path: location.pathname,
+            });
+          }}
+        >
+          Build a dashboard
+        </LinkButton>
+
+        {/* Explore */}
+        {hasExploreRights && (
+          <LinkButton
+            icon="compass"
+            fill="outline"
+            variant="secondary"
+            className={styles.button}
+            href={constructDataSourceExploreUrl(dataSource)}
+            onClick={() => {
+              trackExploreClicked({
+                grafana_version: config.buildInfo.version,
+                datasource_uid: dataSource.uid,
+                plugin_name: dataSource.typeName,
+                path: location.pathname,
+              });
+            }}
+          >
+            Explore
+          </LinkButton>
+        )}
+      </Card.Tags>
+    </Card>
+  );
+}
+
+function DataSourcesListCardSkeleton({ hasExploreRights }: Pick<Props, 'hasExploreRights'>) {
+  const skeletonStyles = useStyles2(getSkeletonStyles);
+  return (
+    <Card>
+      <Card.Heading>
+        <Skeleton width={140} />
+      </Card.Heading>
+      <Card.Figure>
+        <Skeleton width={40} height={40} containerClassName={skeletonStyles.figure} />
+      </Card.Figure>
+      <Card.Meta>
+        <Skeleton width={120} />
+      </Card.Meta>
+      <Card.Tags>
+        <Stack direction="row" gap={2}>
+          <Skeleton height={32} width={179} containerClassName={skeletonStyles.button} />
+
+          {/* Explore */}
+          {hasExploreRights && <Skeleton height={32} width={107} containerClassName={skeletonStyles.button} />}
+        </Stack>
+      </Card.Tags>
+    </Card>
+  );
+}
+
+DataSourcesListCard.Skeleton = DataSourcesListCardSkeleton;
```

Comment:
I haven't done Skeletons myself but is this the right pattern? I thought you'd simply export a separate component instead of an attribute of the original component

Reply:
maybe, maybe not 🤷  is there a downside to doing this i'm not aware of? 🤔 couple of pros:
- skeletons often share the same styles as the base component, so it makes sense that they live together in the same file
- it's easier to know if a component has an existing skeleton
- it's easier to use. no extra imports needed.

i'm gonna present it at frontend-weekly after christmas and if we decide on a different pattern then i can refactor these existing cases 👍

Reply:
I like it! I really have no strong argument against this pattern but it doesn't look like the way you usually export components in react, that's why I ask. 

Let's see what people say in the weekly, so far good enough for me :D